packages/fiber/src/core/index.tsx

```diff
@@ -335,34 +335,36 @@ function Portal({
    *  <Canvas>
    *    {createPortal(...)} */
 
-  const { events, ...rest } = state
   const previousRoot = useStore()
   const [raycaster] = React.useState(() => new THREE.Raycaster())
   const [pointer] = React.useState(() => new THREE.Vector2())
 
+  type OtherState = Omit<RootState, 'size' | 'viewport' | 'internal' | 'performance'>
   const inject = React.useCallback(
-    (state: RootState, injectState?: RootState) => {
-      const intersect: Partial<RootState> = { ...state }
+    (prevRootState: RootState, injectState?: RootState) => {
+      const { events, ...rest } = state
+      const intersect: Partial<RootState> = { ...prevRootState }
 
       if (injectState) {
         // Only the fields of "state" that do not differ from injectState
-        Object.keys(state).forEach((key) => {
+
+        // Some props should be off-limits. Discarding off-limits props from being checked
+        let { size: _, viewport: __, internal: ___, performance: ____, ...otherState } = prevRootState
+        for (let key in otherState) {
           if (
-            // Some props should be off-limits
-            !['size', 'viewport', 'internal', 'performance'].includes(key) &&
-            // Otherwise filter out the props that are different and let the inject layer take precedence
-            state[key as keyof RootState] !== injectState[key as keyof RootState]
```

Comment:
What if we moved the private keys' array into global scope?

Reply:
Do You mean these keys `['size', 'viewport', 'internal', 'performance']` ? I personally don't think that's needed because from what I understand that's very specific to inject function

Reply:
> Not really a fan of destructuring to exclude keys in the above, we're already iterating over keys.

My first thought was since those keys are just going to be ignored anyway so maybe it's the best to exclude them. But if you want to check all the keys than it's fine I'll change it